extensions/funqy/funqy-knative-events/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/knative/events/KnativeEventsBindingRecorder.java

```diff
@@ -55,18 +64,38 @@
 
     public void init() {
         typeTriggers = new HashMap<>();
+        invokersFilters = new HashMap<>();
         objectMapper = getObjectMapper()
                 .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
                 .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true);
         queryMapper = new QueryObjectMapper();
         for (FunctionInvoker invoker : FunctionRecorder.registry.invokers()) {
             Method method = invoker.getMethod();
+            String trigger;
             CloudEventMapping annotation = method.getAnnotation(CloudEventMapping.class);
+            final List<Predicate<CloudEvent>> filter;
             if (annotation != null && !annotation.trigger().isEmpty()) {
-                typeTriggers.put(annotation.trigger(), invoker);
+                trigger = annotation.trigger();
+                filter = filter(invoker.getName(), annotation);
             } else {
-                typeTriggers.put(invoker.getName(), invoker);
+                trigger = invoker.getName();
+                filter = Collections.emptyList();
             }
+            invokersFilters.put(invoker.getName(), filter);
+            typeTriggers.compute(trigger, (k, v) -> {
+                if (v == null) {
+                    v = new ArrayList<>();
+                }
+                // validate if there are no duplicates for the same type (trigger) and defined filters
```

Comment:
I think `duplicates` is not correct anymore. We are checking for sub-set relation.

Reply:
replaced with conflicting